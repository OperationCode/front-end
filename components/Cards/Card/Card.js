import React from 'react';
import { bool, node, string } from 'prop-types';
import classNames from 'classnames';
import { getDataAttributes } from 'common/utils/prop-utils';
import styles from './Card.module.css';

Card.propTypes = {
  children: node.isRequired,
  className: string,
  hasAnimationOnHover: bool,
};

Card.defaultProps = {
  className: undefined,
  hasAnimationOnHover: false,
};

function Card({ children, className, hasAnimationOnHover, ...props }) {
  const customDataAttributes = getDataAttributes(props);

  return (
    <article
      className={classNames(styles.Card, className, { [styles.animatedCard]: hasAnimationOnHover })}
      {...customDataAttributes}
    >
      {children}
    </article>
  );
}

export default Card;
