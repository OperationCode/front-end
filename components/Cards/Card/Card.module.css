.Card {
  align-items: center;
  background-color: var(--white);
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  color: var(--secondary);
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
  margin: 1rem;
  min-height: 100px;
  min-width: 100px;
  padding: 1.5rem;
}

.Card svg {
  fill: var(--secondary);
}

.Card:focus {
  outline: none;
}

.animatedCard {
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  transition: 0.2s box-shadow linear;
}

.animatedCard:focus,
.animatedCard:hover {
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.25);
}
