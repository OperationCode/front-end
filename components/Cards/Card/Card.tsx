import { cx } from 'common/utils/cva';
import { getDataAttributes } from 'common/utils/prop-utils';

export interface CardPropsType {
  children: React.ReactNode;
  className?: string;
  hasAnimationOnHover?: boolean;
}

function Card({ children, className, hasAnimationOnHover, ...props }: CardPropsType) {
  const customDataAttributes = getDataAttributes(props);

  return (
    <article
      className={cx(
        'items-center bg-white [&_svg]:fill-secondary text-secondary flex flex-col flex-nowrap justify-around m-4 min-h-[100px] min-w-[100px] p-6 shadow-md focus-visible:outline-none',
        hasAnimationOnHover &&
          'shadow-sm transition-shadow duration-200 ease-linear hover:shadow-lg focus-visible:shadow-lg',
        className,
      )}
      {...customDataAttributes}
    >
      {children}
    </article>
  );
}

export default Card;
