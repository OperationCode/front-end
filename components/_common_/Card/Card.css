.Card {
  align-items: center;
  background-color: #ffffff;
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
  margin: 1rem;
  min-height: 100px;
  min-width: 100px;
  padding: 1.5rem;
}

.Card:focus {
  outline: none;
}

.animatedCard {
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  transition: 0.2s box-shadow linear;
}

.animatedCard:focus,
.animatedCard:hover {
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.25);
}
