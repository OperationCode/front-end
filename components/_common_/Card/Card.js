import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Card.css';

Card.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasAnimationOnHover: PropTypes.bool,
};

Card.defaultProps = {
  className: '',
  hasAnimationOnHover: false,
};

function Card({ children, className, hasAnimationOnHover }) {
  return (
    <article
      className={classNames(styles.Card, className, { [styles.animatedCard]: hasAnimationOnHover })}
    >
      {children}
    </article>
  );
}

export default Card;
