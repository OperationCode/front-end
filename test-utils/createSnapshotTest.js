/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
