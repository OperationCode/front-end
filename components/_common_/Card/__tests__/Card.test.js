/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Card from '../Card';

describe('Card', () => {
  test('should render with just required props passed', () => {
    createSnapshotTest(<Card>Test</Card>);
  });

  test('should render properly with some props assigned', () => {
    createSnapshotTest(
      <Card className="test-class" hasAnimationOnHover>
        Test
      </Card>,
    );
  });
});
