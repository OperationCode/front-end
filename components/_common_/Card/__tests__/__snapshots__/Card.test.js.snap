// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly with some props assigned 1`] = `
<article
  className="Card test-class animatedCard"
>
  Test
</article>
`;

exports[`Card should render with just required props passed 1`] = `
<article
  className="Card"
>
  Test
</article>
`;
